install/install.go

```diff
@@ -965,3 +971,50 @@ func (k *K8sInstaller) RollbackInstallation(ctx context.Context) {
 		r(ctx)
 	}
 }
+
+func (k *K8sInstaller) InstallWithHelm(ctx context.Context, k8sClient genericclioptions.RESTClientGetter) error {
+	// If --list-versions flag is specified, print available versions and return.
+	if k.params.ListVersions {
+		return k.listVersions()
+	}
+	if err := k.autodetectAndValidate(ctx); err != nil {
+		return err
+	}
+
+	switch k.flavor.Kind {
+	case k8s.KindGKE:
+		// Get Helm values to check if ipv4NativeRoutingCIDR value is specified via a Helm flag.
+		helmValues, err := k.params.HelmOpts.MergeValues(getter.All(cli.New()))
+		if err != nil {
+			return err
+		}
+		if helmValues["ipv4NativeRoutingCIDR"] == nil {
+			cidr, err := k.gkeNativeRoutingCIDR(ctx, k.client.ContextName())
+			if err != nil {
+				k.Log("❌ Unable to auto-detect GKE native routing CIDR. Is \"gcloud\" installed?")
+				k.Log("ℹ️  You can set the native routing CIDR manually with --helm-set ipv4NativeRoutingCIDR=x.x.x.x/x")
+				return err
+			}
+			k.params.IPv4NativeRoutingCIDR = cidr
+		}
+	}
```

Comment:
Instead of duplicating this logic here and in `(*K8sInstaller).Install`, maybe move it to a common function. Or alternatively, drop this method altogether and move the helm-specific logic into `(*K8sInstaller).Install` behind a newly introduced `withHelm bool` parameter?